OpenQA/Isotovideo/CommandHandler.pm

```diff
@@ -53,6 +58,28 @@ has backend_requester => undef;
 # whether the test has already been completed and whether it has died
 has [qw(test_completed test_died)] => 0;
 
+# the time of the last asserted screen
+has [qw(last_check_seconds last_check_microseconds)] => 0;
+
+sub new ($class, @args) {
+    my $self = $class->SUPER::new(@args);
+
+    my $signalhandler = sub ($sig) { $self->_signalhandler($sig) };
+    $SIG{TERM} = $signalhandler;
+    $SIG{INT} = $signalhandler;
+    $SIG{HUP} = $signalhandler;
```

Comment:
I'd extract that into a function, e.g. called `setup_signal_handler` and have the user calling it after constructing the object. (Assigning global variables unconditionally is not what I'd generally expect when invoking a constructor.)